src/app/modules/ibc.rs

```diff
@@ -304,6 +331,18 @@ impl<S: 'static + ProvableStore> Module for Ibc<S> {
         if let Ok(msg) = decode(message.clone()) {
             debug!("Dispatching message: {:?}", msg);
 
+            #[cfg(feature = "val_exec_ctx")]
+            if cfg!(feature = "val_exec_ctx") {
+                val_exec_dispatch(self, msg)?;
+                let events = self
+                    .events
+                    .drain(..)
```

Comment:
Opened https://github.com/cosmos/ibc-rs/issues/416 so that we can get rid of this `drain()`